onnx/backend/test/cpp/driver_test.cc

```diff
@@ -0,0 +1,139 @@
+#include "driver/gtest_utils.h"
+#include "onnx/checker.h"
+#include "onnx/onnxifi.h"
+#include "onnx/onnxifi_loader.h"
+const bool ONNXIFI_BACKEND_USED = false;
+class ONNXCppDriverTest
+    : public testing::TestWithParam<onnx::testing::ProtoTestCase> {
+ protected:
+  std::vector<onnx::testing::ProtoTestData> protos_;
+  onnx::ModelProto model_;
+  void SetUp() override {
+    onnx::testing::ProtoTestCase t = GetParam();
+    protos_ = t.proto_test_data_;
+    model_ = t.model_;
+  }
+  uint64_t GetDescriptorSize(onnxTensorDescriptorV1* t) {
+    uint64_t d_size = 1;
+    for (int i = 0; i < t->dimensions; i++) {
+      d_size *= t->shape[i];
+    }
+    return d_size;
+  }
+  void RunAndVerify(onnxifi_library& lib, onnxBackend& backend) {
+    // Check Model
+    onnx::checker::check_model(model_);
+    // Check Input&Output Tensors
+    onnx::checker::CheckerContext ctx;
+    for (auto proto_test_data : protos_) {
+      for (auto input : proto_test_data.inputs_) {
+        onnx::checker::check_tensor(input, ctx);
+      }
+      for (auto output : proto_test_data.outputs_) {
+        onnx::checker::check_tensor(output, ctx);
+      }
+    }
+    /* TO DO:
+     * This chunk of code is to test the correctness of onnxifi backend.
+     * Since we are not using a real backend, we should wait and not
+     * enable these tests. */
+
+    if (ONNXIFI_BACKEND_USED) {
+      onnxGraph graph;
+      uint32_t weightCount = model_.graph().initializer_size();
+      onnxTensorDescriptorV1 weightDescriptors =
+          onnx::testing::ProtoToOnnxTensorDescriptor(
+              model_.graph().initializer(0));
+
+      EXPECT_EQ(
+          lib.onnxInitGraph(
+              backend,
+              NULL,
+              sizeof(model_),
+              &model_,
+              weightCount,
+              &weightDescriptors,
+              &graph),
+          ONNXIFI_STATUS_SUCCESS);
+
+      for (const auto& proto_test_data : protos_) {
+        std::vector<onnxTensorDescriptorV1> input_descriptor, output_descriptor,
+            result_descriptor;
+        for (const auto& input : proto_test_data.inputs_) {
+          input_descriptor.push_back(
+              onnx::testing::ProtoToOnnxTensorDescriptor(input));
+        }
+        for (auto& output : proto_test_data.outputs_) {
+          output_descriptor.push_back(
+              onnx::testing::ProtoToOnnxTensorDescriptor(output));
+          onnxTensorDescriptorV1 result;
+          result.tag = ONNXIFI_TAG_TENSOR_DESCRIPTOR_V1;
+          result.name = "result";
```

Comment:
since we expect multiple outputs should the name be unique per output?

Reply:
@rdzhabarov I am not sure about the mechanism of this name, must it be unique? Or it should be unique to distinguish. 
Also, these results are only a temporary vars.
@Maratyszcza What do you think?

Reply:
Name in tensor descriptors correspond to tensor names in the model, which must be unique